### PR TITLE
tests: capture GRPC traffic also

### PR DESCRIPTION
**Pull request by justinsb**
- tfprovider: support intercepting GRPC requests
- tests: record grpc requests in our golden tests
- tests: capture golden output for bigtabletable test

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log
Comment (acpana) [inline pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log:1]: This probably needs to be called something else then, right? Like `_grpc.golden.log` ?

Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log:1]: _maybe_ ... the issue is that it's still http traffic, we're just decoding the GRPC requests.  There are some http logs where we intermingle http and GRPC!

Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log:1]: We could maybe rename `_http.log` to `_requests.log` or `_gcp.log` - it's really the log of non-kube requests/responses.

**Comment by justinsb** [comment]
/assign @yuwenma 

This is on the path for reproducing a bigtable issue (#1780)

File: third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport/bigtable_client_factory.go [hunk 1]
```diff
@@ -32,6 +33,10 @@ func (s BigtableClientFactory) NewInstanceAdminClient(project string) (*bigtable
 	opts = append(opts, option.WithTokenSource(s.TokenSource), option.WithUserAgent(s.UserAgent))
 	opts = append(opts, s.gRPCLoggingOptions...)
 
+	if GRPCUnaryClientInterceptor != nil {
```
Comment (yuwenma) [inline third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport/bigtable_client_factory.go:36]: Is this change from the upstream tf repo fetch  or we manually added the logic?

File: pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log
Comment (yuwenma) [reply pkg/test/resourcefixture/testdata/basic/bigtable/v1beta1/bigtabletable/_http.log:1]: I think keeping the name as it is should be good. From the testing level, it doesn't really matter where the resource comes from or talks to. We care more on the resource itself.

**Comment by yuwenma** [comment]
/lgtm
/approve

The fail test looks suspicious. Could you rerun?

File: third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport/bigtable_client_factory.go (hunk 1 above)
Comment (justinsb) [reply third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport/bigtable_client_factory.go:36]: Added manually

**Pull request merged**